src/uct/ib/ud/base/ud_iface.c

```diff
@@ -841,8 +844,10 @@ static void uct_ud_iface_timer(int timer_id, void *arg)
     now = uct_ud_iface_get_async_time(iface);
     ucs_trace_async("iface(%p) slow_timer_sweep: now %lu", iface, now);
     ucs_twheel_sweep(&iface->async.slow_timer, now);
-    uct_ud_iface_async_progress(iface);
-    uct_ud_iface_raise_pending_async_ev(iface);
+
+    if (uct_ud_iface_async_progress(iface) > 0) {
```

Comment:
need to put this inside uct_ud_iface_async_progress(). in case we will also have interrupt mode, not just timer